Fix: adjusted TTS feedback for Google provider

Devankguptaa opened the pull request:
### Summary
- Added condition to prevent repetitive "Let me check on that." TTS feedback.
- Only triggers for Google TTS provider and skips instant/chart functions.

### Motivation
- Avoids unnecessary repetition and improves user experience.



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- New Features
  - Turn-aware voice prompts: announces “Let me check on that.” only once per turn and skips prompts for instant/visual-only actions.
- Bug Fixes
  - Improved reliability by cleaning up session state on leave, cancel, or errors to prevent leaks.
  - More consistent coordination between function activity and voice prompts.
- Chores
  - Enhanced observability: richer tracing with session/client metadata, consistent tagging, and links for easier troubleshooting.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Praksah-ux committed:
voice: fix repetition issue by gating “Let me check on that.” once per turn (Google TTS)
- Ensures cue plays once per turn, skips instant/chart helpers
- Adds client_sid + application_logs attributes
- Cleans up session state on participant leave/cancel

Fix repetition issue in voice agent initialization

badri-singhal commented on app/agents/voice/automatic/__init__.py:
```diff
@@ -200,22 +200,35 @@ async def main():
 
     rtvi = RTVIProcessor(config=RTVIConfig(config=[]))
 
+        from datetime import datetime, timedelta
```
move this to top

Devankguptaa commented:
Thanks! I've moved the datetime import to the top as suggested. The changes are pushed and the PR is updated.

swaroopvarma1 commented on app/agents/voice/automatic/__init__.py:
```diff
@@ -200,22 +201,35 @@ async def main():
-        # Only play the "checking" message if using Google TTS
+        nonlocal last_check_message_time
+        now = datetime.utcnow()
+        cooldown = timedelta(seconds=5)  # at least 5s gap
```
Instead of this let's only make it speak 'let me check on that' only once for a turn

Devankguptaa commented:
Updated as per feedback — replaced cooldown with a per-turn flag.
Now "Let me check on that." will only be spoken once per turn.

swaroopvarma1 commented:
> Updated as per feedback — replaced cooldown with a per-turn flag. Now "Let me check on that." will only be spoken once per turn.

Can you rebase and squash commits

Devankguptaa commented:
I’ve rebased onto release and squashed the commits into a single one. The branch is updated.

Devankguptaa commented:
can you merge my PR request??

swaroopvarma1 commented on app/agents/voice/automatic/__init__.py:
```diff
@@ -200,22 +201,30 @@ async def main():
+
+        if should_speak and not check_message_spoken:
+            await tts.queue_frame(TTSSpeakFrame("Let me check on that."))
+            check_message_spoken = True   # mark spoken for this turn
```
this seems wrong.
After the first function call, let say after 4-5 turns if a function call happens again, this still we be in  locked state and doesn't speak back 'let me check on  that',  which is not intended

Devankguptaa commented:
✅ Updated as per feedback.

- Implemented **per-turn gating** for the cue.  
  - `_current_turn[sid]` increments on `on_turn_started`.  
  - Cue plays only if `_last_check_prompt_turn[sid] != _current_turn[sid]`.

- Scope: **Google TTS only** and **skips instant/chart helpers**  
  (`get_current_time`, `generate_bar_chart`, `generate_line_chart`, `generate_donut_chart`, `generate_single_stat_card`).

- Turn semantics:  
  - Multiple function calls in the **same turn** → cue plays **once**  
  - **Next user turn** → cue is allowed again  
  - **Non-Google TTS** or **instant/chart-only** → cue suppressed

- Session hygiene: clears per-session gating state on **participant leave** and **pipeline cancel**.

Also updated Langfuse tags to use the voice enum’s string value.

Devankguptaa commented:
can you review my PR now  ??

Devankguptaa committed:
Merge branch 'release' into fix/repetition-issue

Devankguptaa commented:
Fix over-speaking: keep owner’s Google-only + instant-skip + random phrase, but gate TTS to once per turn to prevent repetition. can you review my PR??

Devankguptaa commented:
can you review my PR??

swaroopvarma1 commented:
This has to be now handled via function calling Audio sounds
